### PR TITLE
Design on settings page #10

### LINKED ISSUE #10: Menu
The menu should be available on all pages.
If the user is logged in it should be possible to reach a page for user specific information via the menu. The user should also be able to reach the “Connect to charger” page and log out.
If the user is not logged in, the menu should contain elements for the user to reach the login and register page.

ESTIMATE:  5

### PR DESCRIPTION
**Pull request by scamdivine**
Fixes #10 
Fixes the Settings view to look like the given mockups
A summary explaining what changes the PR has made.

- [ ] added logout button
- [ ] Added navigation links to all new pages 
- [ ] Created custom backbutton 

Skärmavbild 2021-09-20 kl  13 56 29

Short summary of what should be tested
- [ ] Make sure the settings view looks good on different screens 
- [ ] test fancy back button

### CODE REVIEW

File: FlexiCharge/FlexiCharge/Views/Settings/SettingsView.swift [hunk 1]
```diff
@@ -8,8 +8,130 @@
...
+                                .foregroundColor(.black)
+                            Spacer()
+                            Image("menu-arrow-black")
+                        }.frame(width: screenWidth * 0.8, height: screenHeight*0.05, alignment: .bottom)
+                    }
+                    Divider()
+                        .frame(height: 1)
+                        .background(Color(red: 0.90, green: 0.90, blue: 0.90))
+                    /*----------Account Settings----------*/
+                    // Change navigationling to correct page once it is created
```
Comment (LucasStrand) [inline FlexiCharge/FlexiCharge/Views/Settings/SettingsView.swift:74]: navigationling 🇨🇳

File: FlexiCharge/FlexiCharge/Views/Settings/SettingsView.swift [hunk 2]
```diff
@@ -8,8 +8,130 @@
...
+                                .foregroundColor(.black)
+                            Spacer()
+                            Image("menu-arrow-black")
+                        }.frame(width: screenWidth * 0.8, height: screenHeight*0.05, alignment: .bottom)
+                    }
+                    Divider()
+                        .frame(height: 1)
+                        .background(Color(red: 0.90, green: 0.90, blue: 0.90))
+                    /*----------Name and Address----------*/
+                    // Change navigationling to correct page once it is created
```
Comment (LucasStrand) [inline FlexiCharge/FlexiCharge/Views/Settings/SettingsView.swift:88]: navigationling 🍚

File: FlexiCharge/FlexiCharge/Views/Settings/SettingsView.swift [hunk 3]
```diff
@@ -8,8 +8,130 @@
...
+                                .foregroundColor(.black)
+                            Spacer()
+                            Image("menu-arrow-black")
+                        }.frame(width: screenWidth * 0.8, height: screenHeight*0.05, alignment: .bottom)
+                    }
+                    Divider()
+                        .frame(height: 1)
+                        .background(Color(red: 0.90, green: 0.90, blue: 0.90))
+                    /*----------About----------*/
+                    // Change navigationling to correct page once it is created
```
Comment (LucasStrand) [inline FlexiCharge/FlexiCharge/Views/Settings/SettingsView.swift:102]: navigationling 🍜

File: FlexiCharge/FlexiCharge/Views/Settings/SettingsView.swift [hunk 4]
```diff
@@ -8,8 +8,130 @@
...
+                    /*----------Invoices----------*/
+                    // Change navigationling to correct page once it is created
+                    NavigationLink(destination: ContentView()) {
+                        HStack {
+                            Text("Invoices")
+                                .font(Font.system(size: 20, design: .default))
+                                .foregroundColor(.black)
+                            Spacer()
+                            Image("menu-arrow-black")
+                        }.frame(width: screenWidth * 0.8, height: screenHeight*0.05, alignment: .bottom)
```
Comment (LucasStrand) [inline FlexiCharge/FlexiCharge/Views/Settings/SettingsView.swift:68]: space 👨🏻‍🚀

File: FlexiCharge/FlexiCharge/Views/Settings/SettingsView.swift [hunk 5]
```diff
@@ -8,8 +8,130 @@
...
+                    /*----------Account Settings----------*/
+                    // Change navigationling to correct page once it is created
+                    NavigationLink(destination: ContentView()) {
+                        HStack {
+                            Text("Account Settings")
+                                .font(Font.system(size: 20, design: .default))
+                                .foregroundColor(.black)
+                            Spacer()
+                            Image("menu-arrow-black")
+                        }.frame(width: screenWidth * 0.8, height: screenHeight*0.05, alignment: .bottom)
```
Comment (LucasStrand) [inline FlexiCharge/FlexiCharge/Views/Settings/SettingsView.swift:82]: 📡 🛰️

File: FlexiCharge/FlexiCharge/Views/Settings/SettingsView.swift [hunk 6]
```diff
@@ -8,8 +8,130 @@
...
+                    /*----------Name and Address----------*/
+                    // Change navigationling to correct page once it is created
+                    NavigationLink(destination: ContentView()) {
+                        HStack {
+                            Text("Name and Address")
+                                .font(Font.system(size: 20, design: .default))
+                                .foregroundColor(.black)
+                            Spacer()
+                            Image("menu-arrow-black")
+                        }.frame(width: screenWidth * 0.8, height: screenHeight*0.05, alignment: .bottom)
```
Comment (LucasStrand) [inline FlexiCharge/FlexiCharge/Views/Settings/SettingsView.swift:96]: 🚀 ⭐ 🌔

File: FlexiCharge/FlexiCharge/Views/Settings/SettingsView.swift [hunk 7]
```diff
@@ -8,8 +8,130 @@
...
+                    /*----------About----------*/
+                    // Change navigationling to correct page once it is created
+                    NavigationLink(destination: ContentView()) {
+                        HStack {
+                            Text("About")
+                                .font(Font.system(size: 20, design: .default))
+                                .foregroundColor(.black)
+                            Spacer()
+                            Image("menu-arrow-black")
+                        }.frame(width: screenWidth * 0.8, height: screenHeight*0.05, alignment: .bottom)
```
Comment (LucasStrand) [inline FlexiCharge/FlexiCharge/Views/Settings/SettingsView.swift:110]: yeah, i think you get it at this point

**Review by LucasStrand — CHANGES_REQUESTED** [review]
There's something off about the **Profile & settings** text along with where the back button is geographically placed on the vertical axis. Tweak it a tad bit further down, I believe, should fix it.
Also, while you're at it, and if you feel like it, you can add the fancy back button accordingly to all views that should have it. I know we have some back buttons that are the un modified one. 
Aaaaaalso, if you decide to have a custom placement for the text, tweak the other title texts as well so we have unity in where the text is placed <3 this is of course applied to font size as well.

Great work otherwise! 🥇

File: FlexiCharge/FlexiCharge/Views/Settings/SettingsView.swift [hunk 8]
```diff
@@ -8,8 +8,130 @@
...
+                    /*----------Charging History----------*/
+                    // Change navigationling to correct page once it is created
+                    NavigationLink(destination: LoginView()) {
+                        HStack {
+                            Text("Charging History")
+                                .font(Font.system(size: 20, design: .default))
+                                .foregroundColor(.black)
+                            Spacer()
+                            Image("menu-arrow-black")
+                        }.frame(width: screenWidth * 0.8, height: screenHeight*0.05, alignment: .bottom)
```
Comment (LucasStrand) [inline FlexiCharge/FlexiCharge/Views/Settings/SettingsView.swift:54]: be consistent with spaces `screenHeight*0.05,`

**Review by LucasStrand — APPROVED** [review]
🦸🏻

**Comment by filipfloden** [comment]
The line underneath each row should not be all the way to the edges. Please fix this.

File: FlexiCharge/FlexiCharge/Views/Settings/SettingsView.swift [hunk 9]
```diff
@@ -8,8 +8,134 @@
...
+                            Text("About")
+                                .font(Font.system(size: 20, design: .default))
+                                .foregroundColor(.black)
+                            Spacer()
+                            Image("menu-arrow-black")
+                        }.frame(width: screenWidth * 0.8, height: screenHeight * 0.05, alignment: .bottom)
+                    }
+                    Divider()
+                        .frame(height: 1)
+                        .background(Color(red: 0.90, green: 0.90, blue: 0.90))
```
Comment (filipfloden) [inline FlexiCharge/FlexiCharge/Views/Settings/SettingsView.swift:118]: Refer to comment above

**Review by filipfloden — CHANGES_REQUESTED** [review]

**Review by filipfloden — APPROVED** [review]
Nice work!

**Pull request merged**